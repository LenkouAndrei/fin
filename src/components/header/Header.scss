@import "../../styles/variables.scss";

.header {
    display: flex;
    justify-content: space-between;
    padding: 10px 20px;
    border-bottom: 3px solid $d-hakki;
    margin-bottom: 0;
    flex: 0;
    background-color: $d-green;
    font-size: 2.4rem;
    font-weight: 700;

    &__date {
        color: lighten($d-hakki, 20);
    }
}

.sum {
    &__spend {
        border-right: 2px solid $d-hakki;
        padding-right: 15px;
        color: $white;
    }
    &__balance {
        padding-left: 15px;
        color: $l-blue;
    }
}