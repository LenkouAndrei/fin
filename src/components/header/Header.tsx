import React from 'react';
import './Header.scss';

interface IHeader {
  spent: number;
  restSum: number;
}

export const Header: React.FC<IHeader> = ({ spent, restSum }: IHeader) => {
  return (
    <header className="header">
      <span className="header__date">Date: {new Date().toDateString()}</span>
      <div className="header__cash-content sum">
        <span className="sum__spend">Spend: {spent}</span>
        <span className="sum__balance">Balance: {restSum}</span>
      </div>
    </header>
  );
};
