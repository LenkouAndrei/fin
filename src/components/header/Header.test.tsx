import { render, screen } from '@testing-library/react';
import { Header } from './Header';

describe('<Header />', () => {
  it('should render Input', () => {
      const { asFragment } = render(<Header spent={20} restSum={15}/>);
      expect(asFragment()).toMatchSnapshot();
  });

  it('should apply passed spent prop', () => {
    const spent = 20;
    render(<Header spent={spent} restSum={15}/>);
    const htmlEl = screen.getByText(`Spend: ${spent}`);
    expect(htmlEl).toBeTruthy();
  });

  it('should apply passed restSum prop', () => {
    const restSum = 20;
    render(<Header spent={15} restSum={restSum}/>);
    const htmlEl = screen.getByText(`Balance: ${restSum}`);
    expect(htmlEl).toBeTruthy();
  });
});
