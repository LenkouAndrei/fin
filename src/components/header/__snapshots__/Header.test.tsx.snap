// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Input 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <span
      class="header__date"
    >
      Date: Sun Oct 17 2021
    </span>
    <div
      class="header__cash-content sum"
    >
      <span
        class="sum__spend"
      >
        Spend: 20
      </span>
      <span
        class="sum__balance"
      >
        Balance: 15
      </span>
    </div>
  </header>
</DocumentFragment>
`;
